src/components/Select/index.tsx

```diff
@@ -37,12 +37,19 @@ const Container = styled.div`
 
 Container.defaultProps = { className: 'select' }
 
+export enum SelectItemType {
+  select = 'select',
+  action = 'action',
+}
+
 type SelectItem = {
   value: string
   name: string
   displayName?: string
   leftAdornment?: ReactNode
   rightAdornment?: ReactNode
+  type?: SelectItemType
+  onClick?: Function
```

Comment:
You can add something called [`Discriminated unions`](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) to make sure there is always an onClick when the type is `action`.

Something like this:
```js
type CommonSelectItem = {
  value: string
  name: string
  displayName?: string
  leftAdornment?: ReactNode
  rightAdornment?: ReactNode
}

type TypeProps = { type: 'select'; onClick?: never } | { type: 'action'; onClick: Function }

type SelectItem = CommonSelectItem & TypeProps
```

Reply:
Can't get this to work, and I can't find a way to default the `type` prop on a `SelectItem`, which makes this a bit annoying because then you need to declare for every item the type, although most of the items are always `select`.